### PR TITLE
Feature #12699: RANK Secondary Sorts

### PR DESCRIPTION
**Pull request by hawkfish**
* Use a WindowTokenTree to implement secondary sorting for RANK/PERCENT_RANK window functions.
* Tokens are dense integer equivalents to sort keys that have the same comparison semantics.
* Tweak the parser to accept `ORDER BY` arguments in zero-argument functions.
* ORDER BY peers to make a better test.

### CODE REVIEW

File: src/function/window/window_token_tree.cpp [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	}
+}
+
+unique_ptr<WindowAggregatorState> WindowTokenTree::GetLocalState() {
+	return make_uniq<WindowTokenTreeLocalState>(*this);
+}
+
+void WindowTokenTree::CleanupSort() {
+	//	Convert the deltas to tokens
+	if (mst64) {
```
Comment (lnkuiper) [inline src/function/window/window_token_tree.cpp:92]: Is it worth converting the `if`/`else` control flow to an `enum`? Would we consider extending this with thinner(`uint16_t`)/wider(`uhugeint_t`) types in the future? If not, I'm fine with keeping this as it is.

File: test/sql/window/test_rank_orderby.test [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# name: test/sql/window/test_rank_orderby.test
+# description: Test argument ordering for RANK and PERCENT_RANK
+# group: [window]
+
+statement ok
+PRAGMA enable_verification
+
+query IIII
```
Comment (lnkuiper) [inline test/sql/window/test_rank_orderby.test:8]: Could we add a larger test? Large enough to trigger parallelism (could be with `PRAGMA verify_parallelism`).

**Review by lnkuiper — CHANGES_REQUESTED** [review]
Looks great! I have a few comments:

File: src/function/window/window_token_tree.cpp (hunk 1 above)
Comment (hawkfish) [reply src/function/window/window_token_tree.cpp:92]: Not in the short term, but if I do, then yes we should switch off the width of the physical type.

File: test/sql/window/test_rank_orderby.test (hunk 2 above)
Comment (hawkfish) [reply test/sql/window/test_rank_orderby.test:8]: Heh. I'll try to think of one, but `RANK` is hard to check because it involves peers.

**Review by lnkuiper — APPROVED** [review]
Nice test. Thanks for the change! LGTM now.

**Pull request merged**